ci: `Release` workflow fails because of missing `devDependency` `jest-environment-jsdom`

```log
> react-abac@0.1.17 test /home/runner/work/react-abac/react-abac
> jest --coverage

● Validation Error:

  Test environment jest-environment-jsdom cannot be found. Make sure the testEnvironment configuration option points to an existing node module.

  Configuration Documentation:
  https://jestjs.io/docs/configuration


As of Jest 2[8](https://github.com/rikhoffbauer/react-abac/runs/6495600101?check_suite_focus=true#step:7:9) "jest-environment-jsdom" is no longer shipped by default, make sure to install it separately.
 ERROR  Test failed. See above for more details.
Error: Process completed with exit code 1.
```

**SHOULD** be fixed by adding `jest-environment-jsdom` as a `devDependency`

--- a/package.json
+++ b/package.json
@@ -140,7 +140,7 @@
         "setupFiles": [
             "<rootDir>/test/setupTests.ts"
         ],
-        "testEnvironment": "jsdom",
+        "testEnvironment": "<rootDir>/test/testEnvironment.js",
         "testRegex": "(.*\\.(spec))\\.(ts|tsx|js)$",
         "transform": {
             "^.+\\.(t|j)s$": "ts-jest"
@@ -178,6 +178,7 @@
         "import-sort-style-module": "6.0.0",
         "jest": "28.1.0",
         "jest-cli": "28.1.0",
+        "jest-environment-jsdom-global": "^3.1.1",
         "lint-staged": "12.4.1",
         "prettier": "2.6.2",
         "prettier-plugin-import-sort": "0.0.7",

--- a/package.json
+++ b/package.json
@@ -178,6 +178,7 @@
         "import-sort-style-module": "6.0.0",
         "jest": "28.1.0",
         "jest-cli": "28.1.0",
+        "jest-environment-jsdom": "^28.1.0",
         "jest-environment-jsdom-global": "^3.1.1",
         "lint-staged": "12.4.1",
         "prettier": "2.6.2",

--- a/test/testEnvironment.js
+++ b/test/testEnvironment.js
@@ -1,0 +1,16 @@
+const Environment = require("jest-environment-jsdom-global")
+
+/**
+ * A custom environment to set the TextEncoder
+ */
+module.exports = class CustomTestEnvironment extends Environment {
+    constructor({ globalConfig, projectConfig }, context) {
+        super({ globalConfig, projectConfig }, context);
+
+        if (typeof this.global.TextEncoder === 'undefined') {
+            const { TextEncoder } = require('util');
+            this.global.TextEncoder = TextEncoder;
+        }
+    }
+};
+

